crates/consensus/src/request.rs

```diff
@@ -0,0 +1,46 @@
+use alloy_eips::{
+    eip6110::DepositRequest,
+    eip7002::WithdrawalRequest,
+    eip7685::{Decodable7685, Eip7685Error, Encodable7685},
+};
+use alloy_rlp::{Decodable, Encodable};
+
+/// Ethereum execution layer requests.
+///
+/// See also [EIP-7685](https://eips.ethereum.org/EIPS/eip-7685).
+pub enum Request {
+    /// An [EIP-6110] deposit request.
+    ///
+    /// [EIP-6110]: https://eips.ethereum.org/EIPS/eip-6110
+    DepositRequest(DepositRequest),
+    /// An [EIP-7002] withdrawal request.
+    ///
+    /// [EIP-7002]: https://eips.ethereum.org/EIPS/eip-7002
+    WithdrawalRequest(WithdrawalRequest),
```

Comment:
do we need helper functions like `is_` `as_`?

Reply:
i like that as library convention